### PR TITLE
BD-263 BD-198

### PR DESCRIPTION
**Pull request by KellieHawks**
BD-263: Needed clarification on how to set up purchase object product_id allowing them to segment. 
BD-198: Requested call out of how revenue is calculated within purchase object.
* Added Event Object as Well as part of API cleanup initiative

# Pull Request/Issue Resolution

**Description of Change:**
> I'm changing..... (could be a link, a new image, a new section, etc.)... because...

Closes #**ISSUE_NUMBER_HERE**

### Is this change associated with a Braze feature/product release?
- [ ] Yes (__Insert Feature Release Date Here__)
- [ ] No

> If yes, please note the date of the feature release.

---
---

<details>
<summary>PR Checklist</summary>
- [ ] Check that all links work!
- [ ] Ensure you have completed [our CLA](https://www.braze.com/docs/cla/).
- [ ] Tag @EmilyNecciai as a reviewer when the your work is _done and ready to be reviewed for merge_.
- [ ] Tag others as Reviewers as necessary.
- [ ] If you have modified any links, be sure to add redirects to `config/nginx.conf.erb`.

</details>

<details>
<summary>Writing & Style Quick Tips</summary>

- Use the formatting shown on the [Styling Test Page](https://github.com/Appboy/braze-docs/blob/develop/_docs/_home/styling_test_page.md) to create unique templates and vehicles for your information.
- Use Action verbs!
- Capitalize proper nouns, unique Braze tools and features, custom Braze channels, and all Header titles.
- Redirect when necessary!
- Always speak to the Braze customer - refer to them as “you”, whether they are an Engineer/Developer, a Marketer, or an Admin.
- On Voice: You should aim to be a Friendly Professor. Imagine, if you will, the smartest, but also kindest professor you had. You know, the one who showed you your potential and tested you, but also went slowly in class so you could take great notes and learn some things? Don't be too peppy, but don't be formal. You’re not selling anything - you’re instructing and informing and enabling.

_Consult the [Docs Text Formatting Guide](https://github.com/Appboy/braze-docs/wiki/Special-Formatting) and the [Docs Writing Style Guide & Best Practices](https://github.com/Appboy/braze-docs/wiki/Writing-Style-Guide-&-Best-Practices) for more details._

</details>

<details>
<summary>ATTENTION: For PR Reviewers</summary>
- [ ] Read our [Reviewing a PR page](https://github.com/Appboy/braze-docs/wiki/Reviewing-a-PR) for more on our reviewing suggestions.
- [ ] Preview all changes in the linked Heroku environment (click `View deployment` button below, then "Docs". A `502` error just means you should refresh until you see the Docs Home page.
</details>

---
---

### CODE REVIEW

File: _docs/_api/objects_filters/event_object.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+## What is an Event Object?
+
+An Event Object is an object that gets passed through the API when a specific event occurs. Events objects are housed in an events array. Each Event Object in the events array represents a single occurrence of a Custom Event by a particular user at the designated time value. The event object has many different fields that allow you to customize by setting and using event properties in messages, data collection, and personalization. 
+
+You can check out how to set up custom events for a specific platform by reading the Platform Integration Guide within the [Developer Guide][1]. You can find this information housed within the "Tracking Custom Events" page under the __analytics tab__ of the various platforms. We have linked several for you.
+
+- Tracking Custom Events: <br>[Android][2]<br>[iOS][3]<br>[Web][4]
+
+For information regarding when you should use a Custom Event vs a Custom Attribute, see our [Best Practices] - User Data Collection documentation.
```
Comment (KellieHawks) [inline _docs/_api/objects_filters/event_object.md]: @EmilyNecciai I have done some searching and I think "Best Practices, User Data Collection" no longer exists. Going to delete "For information regarding when you should use a Custom Event vs a Custom Attribute, see our [Best Practices] - User Data Collection documentation." if you don't think this article exists either.

Comment (EmilyNecciai) [reply _docs/_api/objects_filters/event_object.md]: Agreed - I had moved and divided all of the information here: https://www.braze.com/docs/user_guide/data_and_analytics/user_data_collection/

But I think it's looking for something specific that I don't think we have anymore

**Comment by EmilyNecciai** [comment]
@KellieHawks You'll have to add redirects to make sure links go to the new locations. 

https://github.com/Appboy/braze-docs/wiki/Redirects#new-partialhashtag-redirects

Also, do: 

`git pull origin kh/purchase_spec`

before you make more changes to your branch to pull down the updates I made to the PR - I was fixing a merge issues.

File: _docs/_api/objects_filters/event_object.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  // Setting this flag to true will put the API in "Update Only" mode.
+  // When using a "user_alias", "Update Only" mode is always true.
+  "_update_existing_only" : (optional, boolean)
+}
+```
+
+## Event Properties Object
+Custom events and purchases may have event properties. The “properties” values should be an object where the keys are the property names and the values are the property values. Property names must be non-empty strings less than or equal to 255 characters, with no leading dollar signs. Property values can be integers, floats, booleans, datetimes (as strings in ISO8601 or yyyy-MM-dd'T'HH:mm:ss:SSSZ format), or strings less than or equal to 255 characters.
+
+### Event properties
```
Comment (EmilyNecciai) [inline _docs/_api/objects_filters/event_object.md]: Capitalize "P" in Properties.

File: _docs/_api/objects_filters/event_object.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+Event properties __do not__ persist and aren't saved on a user's profile. These properties can, however, be used to trigger messages and for personalization using Liquid, but __does not__ allow you to segment based on these properties. However, Braze does allow you to "save" these properties for 30 days by turning on this feature flipper to keep these properties alive and useable for message personalization. To turn on this feature in your own app group, contact your customer service manager.
```
Comment (EmilyNecciai) [inline _docs/_api/objects_filters/event_object.md]: Amazing! Love this note.

File: _docs/_api/objects_filters/purchase_object.md [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+```
+- [ISO 4217 Currency Code Wiki][20]
+
+## Purchase Product_ID
+
+Within the purchase object, The `product_id` is an identifier for the purchase, e.g Product Name or Product Category
+- Braze allows you to store more than 5000 `product_id`s in the dashboard.
+- `product_id` max is 255 characters
+
+### product_id naming conventions
```
Comment (EmilyNecciai) [inline _docs/_api/objects_filters/purchase_object.md]: Capitalization!

File: _docs/_api/objects_filters/purchase_object.md [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+At Braze, we offer some general naming conventions for the purchase object `product_id`.
```
Comment (EmilyNecciai) [inline _docs/_api/objects_filters/purchase_object.md]: Missing `` around one of the `priduct_id`s

File: _docs/_api/objects_filters/purchase_object.md [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+- `product_id` max is 255 characters
+
+### product_id naming conventions
+At Braze, we offer some general naming conventions for the purchase object `product_id`. 
+When choosing product_id, Braze suggests using simplistic names such as the product name or product category with the intention of grouping all logged items by this `product_id`. 
+
+## Purchase Properties Object
+Custom events and purchases may have event properties. The “properties” values should be an object where the keys are the property names and the values are the property values. Property names must be non-empty strings less than or equal to 255 characters, with no leading dollar signs. Property values can be integers, floats, booleans, datetimes (as strings in ISO8601 or yyyy-MM-dd'T'HH:mm:ss:SSSZ format), or strings less than or equal to 255 characters.
+
+### Purchase properties
```
Comment (EmilyNecciai) [inline _docs/_api/objects_filters/purchase_object.md]: Capitalize "p"

**Pull request merged**